include/s3select.h

```diff
@@ -2238,10 +2250,49 @@ class base_s3object
       m_where_clause->traverse_and_apply(m_sa, m_s3_select->get_aliases(), m_s3_select->is_json_query());
     }
 
+    for (auto& a : *m_s3_select->get_aliases()->get())
```

Comment:
the `set_base_definitions` contains logic per the JSON-output.
it should reside in a separate code-scope (or a function)

this also simplifies the conditions of that code.